### PR TITLE
Feat/current check to time machine

### LINKED ISSUE #14640: Move current check from ChecksState to TimeMachineState

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #14640

### CODE REVIEW

File: ui/src/alerting/actions/checks.ts [hunk 1]
```diff
@@ -15,19 +15,18 @@ import {
...
   Action as NotificationAction,
 } from 'src/shared/actions/notifications'
+import {Action as TimeMachineAction} from 'src/timeMachine/actions'
 
 // Types
 import {RemoteDataState} from '@influxdata/clockface'
-import {Check, GetState, CheckType} from 'src/types'
+import {Check, GetState} from 'src/types'
+import {check1, check2} from '../constants'
+import {setCheckStatus, setTimeMachineCheck} from 'src/timeMachine/actions'
```
Comment (chnn) [inline ui/src/alerting/actions/checks.ts]: Looks like the organization of these imports is a bit mixed up

File: ui/src/alerting/components/EditCheckEO.tsx [hunk 2]
```diff
@@ -33,9 +32,9 @@ import {
 
 interface DispatchProps {
   updateCheck: typeof updateCheck
-  setCurrentCheck: typeof setCurrentCheck
-  getCurrentCheck: typeof getCurrentCheck
-  updateCurrentCheck: typeof updateCurrentCheck
+  setTimeMachineCheck: typeof setTimeMachineCheck
```
Comment (chnn) [inline ui/src/alerting/components/EditCheckEO.tsx]: We used to have a linting rule that would prevent us from shadowing names in the same scope. Not a big deal but I found it helpful to distinguish between a function of type `T` versus one of type `(dispatch: Dispatch) => T` using an `on` prefix for the connected one (`onSetTimeMachineCheck`).

I don't think the `onSetTimeMachineCheck` style needs to be the one true way, just thought I would share.

File: ui/src/shared/components/CheckPlot.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ const CheckPlot: FunctionComponent<Props> = ({
           <ThresholdMarkers
             key="custom"
             thresholds={thresholds}
-            onSetThresholds={updateCheckThresholds}
+            onSetThresholds={updateTimeMachineCheckThresholds}
```
Comment (chnn) [inline ui/src/shared/components/CheckPlot.tsx]: (For later) Seems like we need some way of making thresholds uneditable when a `CheckPlot` is used in a dashboard cell.

File: ui/src/alerting/components/EditCheckEO.tsx (hunk 2 above)
Comment (ebb-tide) [reply ui/src/alerting/components/EditCheckEO.tsx]: I go back and forth on this- I like to be able to follow a function from its definition to where its called without changing its name, But it is true that I have made the mistake of calling wrong function by forgetting to pass it though dispatch multiple times... so perhaps I should just conform to your suggestion.

File: ui/src/shared/components/CheckPlot.tsx (hunk 3 above)
Comment (ebb-tide) [reply ui/src/shared/components/CheckPlot.tsx]: It is currently technically uneditable, but making it explicit would be better

File: ui/src/alerting/components/EditCheckEO.tsx (hunk 2 above)
Comment (ebb-tide) [reply ui/src/alerting/components/EditCheckEO.tsx]: I'll change this here and use on in the future but I'm not going to go through all of this PR

**Pull request merged**